Tools/Pipeline/MainWindow.cs

```diff
@@ -197,7 +199,10 @@ public void UpdateProperties()
 
         public void OutputAppend(string text)
         {
-            Application.Instance.Invoke(() => buildOutput.WriteLine(text));
+            if(Dispatcher.FromThread(Thread.CurrentThread)==null)
+                Application.Instance.AsyncInvoke(() => buildOutput.WriteLine(text));
```

Comment:
Would it not be ok to just always async invoke?  Is there a big performance hit if we do that?